packages/xo-server/src/proxy-console.mjs

```diff
@@ -19,72 +19,48 @@ export default function proxyConsole(ws, vmConsole, sessionId) {
 
   let closed = false
 
-  const socket = connect(
-    {
-      host: hostname,
-      port: url.port || 443,
-      rejectUnauthorized: false,
+  const onSend = error => {
+    if (error) {
+      log.debug('error sending to the XO client:', { error })
+    }
+  }
 
-      // Support XS <= 6.5 with Node => 12
-      minVersion: 'TLSv1',
+  const socket = new WebSocket(url, ['binary'], {
```

Comment:
We have to discuss about this PR because it does not do what I thought, you are changing the behavior here from a WebSocket ⇄ TCP proxy to a WebSocket ⇄ WebSocket proxy.

Reply:
it seems to work in my tests
also : https://stackoverflow.com/questions/15160739/possible-for-websocket-client-on-browser-to-talk-to-tcp-socket-server 
```
Absolutely! That is the purpose of [websockify](https://github.com/kanaka/websockify).

Of course your WebSocket client application will need to be able to implement the protocol of the TCP server. For example, [noVNC](https://github.com/kanaka/noVNC) is a browser VNC client that implements the RFB protocol and by using [websockify](https://github.com/kanaka/websockify) it can connect to a normal TCP based VNC server.

Disclaimer: I created both websockify and [noVNC](https://github.com/kanaka/noVNC)
```

Reply:
tested with 10 console openned , refreshing, and letting the tab "stall" 30mn before reusing it